packages/framework-integration-tests/integration/cli/cli.project.integration.ts

```diff
@@ -117,52 +128,54 @@ describe('Project', () => {
+    const expectedCartDemoGitIgnore = readFileContent('integration/fixtures/cart-demo/.gitignore')
+    const cartDemoGitIgnoreContent = readFileContent(CART_DEMO_GIT_IGNORE)
     expect(cartDemoGitIgnoreContent).to.equal(expectedCartDemoGitIgnore)
 
-    const expectedCartDemoPretierRc = await readFileContent('integration/fixtures/cart-demo/.prettierrc.yaml')
-    const cartDemoPretierRcContent = await readFileContent(CART_DEMO_PRETTIER_RC)
+    const expectedCartDemoPretierRc = readFileContent('integration/fixtures/cart-demo/.prettierrc.yaml')
+    const cartDemoPretierRcContent = readFileContent(CART_DEMO_PRETTIER_RC)
     expect(cartDemoPretierRcContent).to.equal(expectedCartDemoPretierRc)
 
-    const expectedCartDemoPackageJson = await readFileContent('integration/fixtures/cart-demo/package.json')
-    const cartDemoPackageJsonContent = await readFileContent(CART_DEMO_PACKAGE_JSON)
+    const expectedCartDemoPackageJson = readFileContent('integration/fixtures/cart-demo/package.json')
+    const cartDemoPackageJsonContent = readFileContent(CART_DEMO_PACKAGE_JSON)
     packageJsonAssertions(
       expectedCartDemoPackageJson.replace(PROJECT_NAME_FIXTURE_PLACEHOLDER, projectName),
       cartDemoPackageJsonContent,
       ['dependencies', 'devDependencies']
     )
+    const cartDemoPackageJsonObject = JSON.parse(cartDemoPackageJsonContent)
+    expect(cartDemoPackageJsonObject['dependencies']['@boostercloud/framework-core']).to.equal(`^${BOOSTER_VERSION}`)
+    expect(cartDemoPackageJsonObject['dependencies']['@boostercloud/framework-types']).to.equal(`^${BOOSTER_VERSION}`)
+    expect(cartDemoPackageJsonObject['devDependencies']['@boostercloud/cli']).to.equal(`^${BOOSTER_VERSION}`)
 
-    const expectedCartDemoTsConfigEslint = await readFileContent('integration/fixtures/cart-demo/tsconfig.eslint.json')
-    const cartDemoTsConfigEslintContent = await readFileContent(CART_DEMO_TS_CONFIG_ESLINT)
+    const expectedCartDemoTsConfigEslint = readFileContent('integration/fixtures/cart-demo/tsconfig.eslint.json')
+    const cartDemoTsConfigEslintContent = readFileContent(CART_DEMO_TS_CONFIG_ESLINT)
     expect(cartDemoTsConfigEslintContent).to.equal(expectedCartDemoTsConfigEslint)
 
-    const expectedCartDemoTsConfig = await readFileContent('integration/fixtures/cart-demo/tsconfig.json')
-    const cartDemoTsConfigContent = await readFileContent(CART_DEMO_TS_CONFIG)
+    const expectedCartDemoTsConfig = readFileContent('integration/fixtures/cart-demo/tsconfig.json')
+    const cartDemoTsConfigContent = readFileContent(CART_DEMO_TS_CONFIG)
     expect(cartDemoTsConfigContent).to.equal(expectedCartDemoTsConfig)
 
-    process.chdir(projectName)
-    await exec('npm run compile')
-    await exec('npm run lint:check')
-    process.chdir('..')
+    await expect(exec('npm run compile', { cwd: projectName })).to.be.eventually.fulfilled
+    await expect(exec('npm run lint:check', { cwd: projectName })).to.be.eventually.fulfilled
```

Comment:
Fun thing here: when one of these execs failed, the `process.chdir('..')` was skipped, and it made the next tests to write more files in the wrong dir, making the execs fail again in a complete hell of recursive failures that ended up with a ton of new files spread all over the place, and making the final recursive rmdirs fail. Thanks God we have git or I would have been still deleting files by hand 🤣

Reply:
WHAT!? 💥